src/Analyzers/CSharp/Analyzers/UsePatternMatching/CSharpUseNotPatternDiagnosticAnalyzer.cs

```diff
@@ -64,28 +64,46 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
             if (!styleOption.Value)
                 return;
 
-            // Look for the form: !(x is Y y)
-            if (!(node is PrefixUnaryExpressionSyntax
+            // Look for the form: !(x is Y y) and !(x is const)
+            if (node is PrefixUnaryExpressionSyntax
```

Comment:
i'm not a huge fan of all the redundancy here.  i would just make this a simple first check that checks for a `!(...)` and bails otherwise.  then we can just have the simple dedicated code that checks the inner expression.